### ISSUE TITLE
some api modules are not working

### ISSUE THREAD

**Issue by ketan1829** [joaomirandasa/whats2api#10]
Here is my review

1. Current location is not been fetched
2. cannot send current location,send file

**Comment by impozzible** [comment]
I am able to send file

**Comment by joaomirandasa** [comment]
Which routes is not working as expected?

**Comment by ketan1829** [comment]
How to send and receive location, it's not working

I'm impressed by your work I'll surely donate you to improve your project


Thanks for your reply

**Comment by impozzible** [comment]
Oh yeah that's true, the lat and long parameters are not defined inside WhatsApp.js

**Comment by impozzible** [comment]
Locate Whatsapp.js inside modules folder
replace the SANITIZE_MSG with this and it should work
```
var SANITIZE_MSG = function(instanceID,data) {
  return JSON.stringify({
    messages: [{ 
      id: data.id,
      body: data.body,
      fromMe: false,
      self: 0,
      isForwarded: data.isForwarded,
      author: data.from,
      time: data.t,
      lat: data.lat,
      lng: data.lng,
      chatId: data.chat.id,
      type: data.type,
      senderName: data.sender.verifiedName,
      caption: (data.caption ? data.caption : null),
      quotedMsgBody: (data.quotedMsgObj ? data.quotedMsgObj : null),
      chatName: data.sender.formattedName 
    }],
    instanceId: instanceID
  });
};
```

**Comment by ketan1829** [comment]
Thanks for info sir!

i feel glad i found you

you can check this project https://github.com/open-wa/wa-automate-nodejs

which might be helpful for you for improving whats2api


ThankYou

Ketan

**Comment by joaomirandasa** [comment]
Updates are promoted today for version 0.0.6, please pull and check.

**Comment by myfrizqi** [comment]
server API modules (readInstance, reloadServer, etc) not working

**Comment by ketan1829** [comment]
QR Code is not refreshing every 30 secs, and no event is triggered after
authentication of WhatsApp,the qr code is still there!

**Comment by Animeliza** [comment]
Bom dia, João o QR Code não esta funcionando através da WEB e também fiz uma pequena alteração no js whatsapp.js para que qd a msg seja obtida ele marque como lida se fiz no lugar errado me desculpe não sou programador node.

WHATS_API.prototype.PROCESS_MESSAGE = function(data){
  var that = this;
  var SANITIZED = SANITIZE_MSG(that.INSTANCE,data);
  
  **that.CONNECTION.sendSeen(data.from);**
  
  request({
    method: 'POST',
    url:  that.WEBHOOK,
    headers: { 'Content-Type': 'application/json' },
    body: SANITIZED
  }, function(err, response, body){
    if(err){
      ERROR_CATCHER(err);
    } else {
      if(response.statusCode != 200){
        ERROR_CATCHER("Status Code error: "+response.statusCode,response);
      } else {
        console.log(SANITIZED);
      }
    }
  });
};

**Comment by ayushpsc** [comment]
QR code was not getting refreshed however once I scanned it, after 2 to 3 attempts, I got the message in terminal saying whatsapp is ready. 😀😀😀😀
Tested routes via postman and it was working except for some routes for which I will raise a separate issue.

**Comment by joaomirandasa** [comment]
I made some changes on web socket, when qrcode is accepted now you will see an positive check and alert on page to be informed about connection done.

Implemented on 0.0.8
Please check new version of [Postman Collection](https://documenter.getpostman.com/view/7437195/Szf82Twx?version=latest) too
